registry/storage/tagstore.go

```diff
@@ -145,26 +149,45 @@ func (ts *tagStore) Lookup(ctx context.Context, desc distribution.Descriptor) ([
 		return nil, err
 	}
 
-	var tags []string
+	g := errgroup.Group{}
```

Comment:
If any tag lookups fail the whole result is discarded, so any work performed after the first failure is a waste of time and resources. Cancelation is trivial with errgroup when the work accepts a context, so we may as well implement it.
```suggestion
	g, ctx := errgroup.WithContext(ctx)
```

Also, break the loop early.
```go
for _, tag := range allTags {
	if ctx.Err() != nil {
		break
	}
	...
}
```